### PR TITLE
Access /proc via $HOST_ROOT

### PR DESCRIPTION
**Pull request by alban**
In preparation for https://github.com/inspektor-gadget/inspektor-gadget/issues/1658

### CODE REVIEW

File: gadget-container/hooks/oci/main.go [hunk 1]
```diff
@@ -48,6 +48,12 @@ func init() {
 	flag.StringVar(&hook, "hook", "", "OCI hook: prestart or poststop")
 }
 
+var hostRoot string
+
+func init() {
+	hostRoot = os.Getenv("HOST_ROOT")
+}
+
```
Comment (flyth) [inline gadget-container/hooks/oci/main.go:56]: Maybe put this into a (new) common library to avoid code repetition?

File: pkg/host/host-root.go
Comment (mauriciovasquezbernal) [inline pkg/host/host-root.go:1]: What about putting this under pkg/utils/host. Or perhaps under internal? (I think this is not useful for external users and pkg/host is too generic).

File: pkg/container-collection/options.go [hunk 2]
```diff
@@ -24,6 +24,8 @@ import (
 	"sync"
 	"time"
 
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/host"
+
```
Comment (mauriciovasquezbernal) [inline pkg/container-collection/options.go:28]: Move this import below with other inspektor-gadget packages.

File: pkg/gadgets/top/ebpf/tracer/tracer.go [hunk 3]
```diff
@@ -27,6 +27,8 @@ import (
 	"strings"
 	"time"
 
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/host"
+
```
Comment (mauriciovasquezbernal) [inline pkg/gadgets/top/ebpf/tracer/tracer.go:31]: ditto.

File: pkg/gadgets/top/ebpf/tracer/tracer.go [hunk 4]
```diff
@@ -196,7 +198,7 @@ func getPidMapFromProcFs() (map[uint32][]*types.Process, error) {
 }
 
 func getMemoryUsage(m *ebpf.Map) (uint64, error) {
-	f, err := os.Open(fmt.Sprintf("/proc/self/fdinfo/%d", m.FD()))
+	f, err := os.Open(fmt.Sprintf("%s/proc/self/fdinfo/%d", host.HostRoot, m.FD()))
```
Comment (mauriciovasquezbernal) [inline pkg/gadgets/top/ebpf/tracer/tracer.go:201]: Some places use `filepath.Join` and others `fmt.Sprintf`. Can we uniform it? (Given that this code is linux specific, can we only use fmt.Sprintf?)

File: pkg/host/host-root.go
Comment (alban) [reply pkg/host/host-root.go:1]: Moved to pkg/utils/host.

File: pkg/gadgets/top/ebpf/tracer/tracer.go (hunk 4 above)
Comment (alban) [reply pkg/gadgets/top/ebpf/tracer/tracer.go]: I switched to filepath.Join, I find it more clear for semantics.

**Review by mauriciovasquezbernal — APPROVED** [review]
LGTM!

nit: I'd be nice to have a small unit test for pkg/utils/host/host.go, but it's up to you.

**Comment by alban** [comment]
Thanks! I will merge as soon as I understand the issue behind https://github.com/inspektor-gadget/inspektor-gadget/pull/1665#issuecomment-1561374258.

**Pull request merged**